### PR TITLE
if accepted this commit will complete ncnews7

### PR DESCRIPTION
**Pull request by Paulos-se**

### CODE REVIEW

**Review by Angaca — CHANGES_REQUESTED** [review]
It works very smoothly and I really like it, the only change you need is that you should allow the user to only like/dislike once and only if I change my mind and remove that vote re like/dislike

**Review by Angaca — CHANGES_REQUESTED** [review]
Still some changes are needed, now I can correctly only vote once but I cannot change my mind, remove my casted vote and re-do it if I like. Plus is not a nice user experience that the button is clickable but does nothing. 
What should happen is having the button press and you increment, re press the button to remove it but then I should be able to press it again, the opposite button should be disabled while my vote is casted and reactivated when I remove my vote

**Review by HeatherEver — COMMENTED** [review]
Looks good but I would maybe add once more piece of functionality - when the like button is pressed and the like goes up - if we press the like button again to 'deselect' it i would be nice if the votes went down. 
This features can just be added into the if (liked) statement at the top of your functions to take away that vote
same for voting down please - then it's complete!

**Review by Angaca — CHANGES_REQUESTED** [review]
Love it visually, there is a bug to fix, if I keep changing them the website crash :(

File: src/components/UpdateVote.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import axios from "axios";
+import { useState } from "react";
+
+function UpdateVote({ vote, setVote, setArticle, article }) {
+  const [liked, setLiked] = useState(false);
+  const [unLiked, setUnLiked] = useState(false);
+  const [likeDisable, likeSetDisable] = useState(false);
+  const [unLikeDisable, unLikeSetDisable] = useState(false);
```
Comment (douglashellowell) [inline src/components/UpdateVote.jsx:8]: these booleans seem to be changing at the same time to the same value - how is `liked` and `likeDisable` different?

File: src/components/UpdateVote.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          `https://nc-news-pa.herokuapp.com/api/articles/${article.article_id}`,
+          { inc_votes: -1 }
+        )
+        .then((res) => {
+          setVote(res.data.votes);
+        });
+
+      setVote((current) => {
+        let newVote = current - 1;
+
```
Comment (douglashellowell) [inline src/components/UpdateVote.jsx:20]: a lot of repeated code here - try to keep your code DRY

If the differences in these functions is `+1` or `-1` then consider passing it in as a variable and having one function that handles both

**Comment by douglashellowell** [comment]
[link to code](https://github.dev/Paulos-se/nc-news/blob/91bbd0d7210352226f867b6e28a05b05b931010e/src/components/SingleArticle.jsx#L29-L30)
I'm not sure if you want to include `isLoading` in the dependancy array here
You only want to fetch the article again if the `article_id` changes!

**Comment by douglashellowell** [comment]
There's some inconsistancy in how your buttons look when they are clickable or disabled - see video

https://user-images.githubusercontent.com/53038423/181267896-568eb98c-4b32-4a69-a58b-7d476eb8c384.mov

**Comment by douglashellowell** [comment]
Final bit of feedback from me:

At the moment you are rendering the votes on the page but they look like just a random number!

put a symbol or some text near the number to indicate what it represents

**Comment by douglashellowell** [comment]
CSS on those buttons is much better 🙌

The code should still be tidied up here - you don't need both your `upVote` and `downVote` functions

after that it's ready to merge! 🚀

**Pull request merged**